File: packages/python/src/constellation_ded_sdk/core/document.py
"""Document hashing using SHA-256."""

from constellation_sdk import hash_bytes


def hash_document(content: str | bytes) -> str:
    """Hash document content using SHA-256, returning a hex string.

    This is the standard way to produce a ``document_ref`` for a FingerprintValue.

    Args:
        content: Document content as UTF-8 string or raw bytes.

    Returns:
        Hex-encoded SHA-256 hash (64 characters).
    """
    data = content.encode("utf-8") if isinstance(content, str) else content
    return hash_bytes(data).value

File: packages/typescript/src/core/document.ts
import { hashBytes } from '@constellation-network/metagraph-sdk';

/**
 * Hash document content using SHA-256, returning a hex string.
 *
 * This is the standard way to produce a `documentRef` for a FingerprintValue.
 * Accepts either a UTF-8 string or raw bytes (Uint8Array).
 *
 * @param content - Document content as string or bytes
 * @returns Hex-encoded SHA-256 hash (64 characters)
 */
export function hashDocument(content: string | Uint8Array): string {
  const bytes =
    typeof content === 'string'
      ? new TextEncoder().encode(content)
      : content;
  return hashBytes(bytes).value;
}
